mlrun/runtimes/sparkjob/spark2job.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2018 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from mlrun.config import config
+
+from ...utils import update_in
+from .abstract import AbstractSparkDefaults, AbstractSparkJobSpec, AbstractSparkRuntime
+
+
+class Spark2Defaults(AbstractSparkDefaults):
+    igz_deps = {
+        "jars": [
+            "/spark/v3io-libs/v3io-hcfs_2.11.jar",
+            "/spark/v3io-libs/v3io-spark2-streaming_2.11.jar",
+            "/spark/v3io-libs/v3io-spark2-object-dataframe_2.11.jar",
+            "/igz/java/libs/scala-library-2.11.12.jar",
+        ],
+        "files": ["/igz/java/libs/v3io-pyspark.zip"],
+    }
+    spark_version = "2.4.5"
```

Comment:
Wouldn't it be more elegant to define abstract methods on the abstract class - `_get_igz_deps` and `_get_spark_version` so it will clear what need to be implemented when overriding the class ?